datasets/qed/qed.py

```diff
@@ -0,0 +1,137 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""QED: A Dataset for Explanations in Question Answering"""
+
+from __future__ import absolute_import, division, print_function
+
+import json
+
+import datasets
+
+
+_CITATION = """\
+@misc{lamm2020qed,
+    title={QED: A Framework and Dataset for Explanations in Question Answering},
+    author={Matthew Lamm and Jennimaria Palomaki and Chris Alberti and Daniel Andor and Eunsol Choi and Livio Baldini Soares and Michael Collins},
+    year={2020},
+    eprint={2009.06354},
+    archivePrefix={arXiv},
+    primaryClass={cs.CL}
+}
+"""
+
+_DESCRIPTION = """\
+QED, is a linguistically informed, extensible framework for explanations in question answering. \
+A QED explanation specifies the relationship between a question and answer according to formal semantic notions \
+such as referential equality, sentencehood, and entailment. It is an expertannotated dataset of QED explanations \
+built upon a subset of the Google Natural Questions dataset.
+"""
+
+_HOMEPAGE = "https://github.com/google-research-datasets/QED"
+
+_BASE_URL = "https://raw.githubusercontent.com/google-research-datasets/QED/master/"
+_URLS = {
+    "train": _BASE_URL + "qed-train.jsonlines",
+    "dev": _BASE_URL + "qed-dev.jsonlines",
+}
+
+
+class Qed(datasets.GeneratorBasedBuilder):
+    """QED: A Dataset for Explanations in Question Answering"""
+
+    VERSION = datasets.Version("1.0.0")
+    BUILDER_CONFIGS = [
+        datasets.BuilderConfig(name="qed", version=datasets.Version("1.0.0")),
+    ]
+
+    def _info(self):
+        span_features = {
+            "start": datasets.Value("int32"),
+            "end": datasets.Value("int32"),
+            "string": datasets.Value("string"),
+        }
+        reference_features = {
+            "start": datasets.Value("int32"),
+            "end": datasets.Value("int32"),
+            "bridge": datasets.Value("bool_"),
+            "string": datasets.Value("string"),
+        }
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=datasets.Features(
+                {
+                    "example_id": datasets.Value("int64"),
+                    "title_text": datasets.Value("string"),
+                    "url": datasets.Value("string"),
+                    "question": datasets.Value("string"),
+                    "paragraph_text": datasets.Value("string"),
+                    "sentence_starts": datasets.Sequence(datasets.Value("int32")),
+                    "original_nq_answers": [[span_features]],
```

Comment:
why is it nested as lis of list of spans ?

Reply:
I went with the original structure, but looked at the data again, and yes inner list is unnecessary

Reply:
Let's simplify that one then